src/resolve/resolved-llmnr.c

```diff
@@ -134,58 +134,90 @@ int manager_llmnr_ipv4_udp_fd(Manager *m) {
         if (m->llmnr_ipv4_udp_fd >= 0)
                 return m->llmnr_ipv4_udp_fd;
 
-        m->llmnr_ipv4_udp_fd = socket(AF_INET, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
-        if (m->llmnr_ipv4_udp_fd < 0)
+        r = socket(AF_INET, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
+        if (r < 0) {
+                log_error_errno(r, "LLMNR-IPv4(udp): Failed to create socket: %m");
```

Comment:
Ooops, I didn't notice that before. Much of the error handling is wrong. It should be more correct to do:
```C
return log_error_errno(errno, "LLMNR-IPv4(udp): Failed to create socket: %m");
```
(We don't log the -1 that `socket` returns. It is useless.)
But there's also no need to set `r`. So you can just do:

```C
m->llmnr_ipv4_udp_fd = socket(AF_INET, SOCK_DGRAM|SOCK_CLOEXEC|SOCK_NONBLOCK, 0);
if (m->llmnr_ipv4_udp_fd < 0)
     return log_error_errno(errno, "LLMNR-IPv4(udp): Failed to create socket: %m");
```